pkgs/build-support/rust/import-cargo-lock.nix

```diff
@@ -0,0 +1,104 @@
+{ fetchgit, fetchurl, lib, runCommand }:
```

Comment:
While I like this approach it is a bit of an issue with cargo workspaces that contain more than one crate. I often just hacki in an override mechanism to select output dirs based on a crate name an maintain that.

@alyssais did some work on that issue in a branch of theirs. Perhaps that approach might fit in here.

Reply:
I think my latest changes should resolve this issue (if I understand it correctly). E.g. using the `rand` crate works, even though it is part of a workspace and uses other dependencies from the workspace (see the added tests).